test/expect/TestSparseCSRCPU.test_sparse_csr_print_cpu.expect

```diff
@@ -9,7 +9,7 @@ tensor(crow_indices=tensor([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10]),
        col_indices=tensor([5, 1, 6, 5, 6, 4, 2, 5, 5, 9]),
        values=tensor([ 0.5674,  0.1261,  0.5497,  0.6416, -0.4414,  0.3634,
                       -0.4327,  0.3135, -0.5225,  0.4626]), size=(10, 10),
-       nnz=10)
+       nnz=10, layout=torch.sparse_csr)
```

Comment:
It looks to me like this is because you are no longer clearing the contents of the tensors, so their previous contents leak through in the tests.

It seems to me like there are three possible approaches here:
1. The tests are wrong, we should never be printing a sparse tensor immediately after `resize_and_clear_`, please fix this. (Clear in this case means that "the tensor is an indeterminate state"; this is better for performance)
2. `resize_and_clear_` is wrong, you need to actually *clear* the tensor after the resizing. (This is worse for performance, because the clear might be redundant in some contexts)
3. `resize_and_clear_` as a function is just entirely philosophically wrong and we shouldn't support it in the API

Reply:
Related comment in https://github.com/pytorch/pytorch/pull/57274#discussion_r630262891 : creating a new CSR does not require calling `resize_and_clear_`.

Reply:
https://github.com/pytorch/pytorch/pull/57274 implements checking CSR invariants that will avoid the above-mentioned bug of creating invalid CSR tensors.

Reply:
> It looks to me like this is because you are no longer clearing the contents of the tensors, so their previous contents leak through in the tests.
> 
> It seems to me like there are three possible approaches here:
> 
> 1. The tests are wrong, we should never be printing a sparse tensor immediately after `resize_and_clear_`, please fix this. (Clear in this case means that "the tensor is an indeterminate state"; this is better for performance)
> 2. `resize_and_clear_` is wrong, you need to actually _clear_ the tensor after the resizing. (This is worse for performance, because the clear might be redundant in some contexts)
> 3. `resize_and_clear_` as a function is just entirely philosophically wrong and we shouldn't support it in the API


Agree with @pearu  that the creation of  new CSR does not require calling `resize_and_clear_`. Agree with point number 3, `resize_and_clear_` as a function is wrong and we shouldn't support it in the API as it was not used anywhere except when a new CSR tensor is created. So we can eliminate the call to `resize_and_clear_`, for a new tensor there would be nothing to resize nor clear, and `set_members_tensors` will reset the members anyway.  

I will close this PR.  PR https://github.com/pytorch/pytorch/pull/57274 solves this and the checking CSR invariants.